package/src/index/serialize.rs

```diff
@@ -0,0 +1,119 @@
+//! Defines the serialization format for the package index.
+
+use std::collections::BTreeMap;
+
+use nickel_lang_core::identifier::Ident;
+use serde::{Deserialize, Serialize};
+
+use crate::{
+    index,
+    version::{SemVer, VersionReq},
+    IndexDependency,
+};
+
+use super::{Package, PreciseId};
+
+/// Defines the serialization format for `IndexDependency` in the package index.
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub struct IndexDependencyFormat {
```

Comment:
I suspect you have good reasons to not (de)serialize `IndexDependency` directly and use an indirection instead, but out of curiosity, what are they?

Reply:
I may have gone overboard with this pattern, but I originally had a bunch of trouble caused by `ManifestFile` defining both the deserialization of manifests from the nickel terms and the representation for working with manifests internally. Separating out `ManifestFileFormat` made that all better, so I've been imitating the same pattern elsewhere.